crates/proto/src/domain/notes.rs

```diff
@@ -0,0 +1,33 @@
+use miden_objects::{
+    notes::{NoteMetadata, NoteTag, NoteType},
+    Felt,
+};
+
+use crate::errors::{ConversionError, MissingFieldHelper};
+
+impl TryFrom<crate::generated::note::NoteMetadata> for NoteMetadata {
+    type Error = ConversionError;
+
+    fn try_from(value: crate::generated::note::NoteMetadata) -> Result<Self, Self::Error> {
+        let sender = value
+            .sender
+            .ok_or_else(|| crate::generated::note::NoteMetadata::missing_field("Sender"))?
+            .try_into()?;
+        let note_type = NoteType::try_from(value.note_type as u64)?;
+        let tag = NoteTag::from(value.tag);
+        let aux = Felt::new(value.aux);
```

Comment:
We should probably use `try_from()` here as well as this will silently truncate `u64` range to field element range.